parse.py

```diff
@@ -5,24 +5,43 @@
 import docopt
 import numpy as np
 import pickle
+from collections import defaultdict
 
 import cddb
 import config
 from utils import *
 
 import mwapi
 import mwparserfromhell
-import nltk
+import nltk.data
+from nltk.tokenize import word_tokenize
 
 from keras.models import load_model
 from keras.preprocessing.sequence import pad_sequences
 from keras import backend as K
 K.clear_session()
-K.set_session(K.tf.Session(config=K.tf.ConfigProto(intra_op_parallelism_threads=10, inter_op_parallelism_threads=10)))
+K.set_session(K.tf.Session(config=K.tf.ConfigProto(
+    intra_op_parallelism_threads=10, inter_op_parallelism_threads=10)))
 
 cfg = config.get_localized_config()
 WIKIPEDIA_BASE_URL = 'https://' + cfg.wikipedia_domain
 
+# Tweak the NLTK's pre-trained English sentence tokenizer
+# to recognize more abbreviations
+# https://stackoverflow.com/questions/14095971/how-to-tweak-the-nltk-sentence-tokenizer?
+# The list is created by examining the frequency used in
+# ~9k random sampled articles in Wikipedia
+extra_abbreviations = ['pp', 'no', 'vol', 'ed', 'al', 'e.g', 'etc', 'i.e',
+    'pg', 'dr', 'mr', 'mrs', 'ms', 'vs', 'prof', 'inc', 'incl', 'u.s', 'st',
+    'trans', 'ex']
+sent_detector = nltk.data.load('tokenizers/punkt/english.pickle')
+sent_detector._params.abbrev_types.update(extra_abbreviations)
+
+# Wiki markups to be detected for broken sentences:
+# reference tags, templates {{}}, wikilinks [[]], parentheses (),
+# quotation marks "", italics text ''''
+markup_regex = "(<ref)|({{)|(\[\[)|(\()|(</ref>)|(/>)|(}})|(\]\])|(\))|(\")|('{2})"
```

Comment:
This is probably fine for now, but just fyi - remember `_fast_parse` in Citation Hunt? In it, we are [using the tokenizer](https://github.com/eggpi/citationhunt/blob/master/snippet_parser/core.py#L78) from mwparserfromhell to detect certain tokens like `{{` without having to craft and maintain a regular expression in Citation Hunt itself.

For example, rather than trying to match for `{{`, we'd look for a token of type TemplateOpen, and let mwparserfromhell worry about how to identify the `{{` in the text.

Anyway, no need to do anything like that now, but if you find yourself troubleshooting this regular expression often, it probably makes sense to go that other way.

Reply:
That's cool!! I didn't think of using the tokenizer from mwparserfromhell.. thanks for reminding me :D